module/zfs/spa_errlog.c

```diff
@@ -135,10 +140,6 @@ name_to_bookmark(char *buf, zbookmark_phys_t *zb)
 }
 
 #ifdef _KERNEL
-static int check_clones(spa_t *spa, uint64_t zap_clone, uint64_t snap_count,
-    uint64_t *snap_obj_array, zbookmark_err_phys_t *zep, void* uaddr,
-    uint64_t *count);
```

Comment:
Did you happen to try declaring this as a `static inline` function instead?  I'd be curious to know if explicitly requesting this results in any saving.  Since this function is already `static` and only called in one location, gcc may have already gone and inlined this function depending on the optimization level.

https://gcc.gnu.org/onlinedocs/gcc/Inline.html

Reply:
Yes, I tried out that first. In that case, the Linux stack tracer reports:
```
 70)     1584     104   check_clones+0x142/0x1d0 [zfs]
 71)     1480     192   check_filesystem+0x319/0x400 [zfs]
 72)     1288     104   check_clones+0x142/0x1d0 [zfs]
 73)     1184     192   check_filesystem+0x319/0x400 [zfs]
 74)      992     104   check_clones+0x142/0x1d0 [zfs]
 75)      888     192   check_filesystem+0x319/0x400 [zfs]
 76)      696      64   process_error_list+0xfa/0x110 [zfs]
 77)      632      56   spa_get_errlog+0xe7/0x2a0 [zfs]
```

That report suggests (falsely?) that the cumulative stack size of `check_filesystem()` is 192 (itself) + 104 (check_clones) = 296 bytes. Whereas, if I integrate `check_clones()` manually into `check_filesystem()` it reports:

```
 41)     1648     192   check_filesystem+0x471/0x5b0 [zfs]
 42)     1456     192   check_filesystem+0x471/0x5b0 [zfs]
 43)     1264     192   check_filesystem+0x471/0x5b0 [zfs]
 44)     1072     192   check_filesystem+0x471/0x5b0 [zfs]
 45)      880     192   check_filesystem+0x471/0x5b0 [zfs]
 46)      688      64   process_error_list+0xfa/0x110 [zfs]
```

showing that `check_filesystem()` has a stack size of 192 bytes. I found the report of the tracer with `static inline` confusing and opted for the second option.

Reply:
It looks like `check_clones()` actually wasn't inlined in the first case (it is optional), so I agree the second option is the way to go.  Thanks for checking.